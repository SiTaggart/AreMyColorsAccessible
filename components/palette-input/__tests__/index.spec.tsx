/* eslint-env jest */
/// <reference types="jest" />
import { expect } from '@jest/globals';

import { render } from '@testing-library/react';
import React from 'react';
import { PaletteInput } from '..';

describe('PaletteInput', (): void => {
  const mockOnColorAdd: jest.Mock = jest.fn();

  it('renders without crashing', (): void => {
    const { asFragment } = render(<PaletteInput onColorAdd={mockOnColorAdd} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render an error message when passed', (): void => {
    const { asFragment } = render(
      <PaletteInput errorMessage="I'm an error" onColorAdd={mockOnColorAdd} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
