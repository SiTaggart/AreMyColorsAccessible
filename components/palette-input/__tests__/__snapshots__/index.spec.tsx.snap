// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteInput renders without crashing 1`] = `
<DocumentFragment>
  .emotion-0 {
  text-align: center;
}

.emotion-2 {
  font-weight: 700;
  line-height: 1;
  margin: 3rem 0;
}

.emotion-4 {
  display: inline-block;
  font-size: 2.5rem;
  margin-bottom: 0.5rem;
  width: 100%;
}

.emotion-6 {
  background: rgba(255, 255, 255, 0.1);
  border: solid 1px;
  border-radius: 3px;
  font-size: 1.2rem;
  padding: 0.5rem 1rem;
  margin-bottom: 0;
  width: 100%;
}

<form
    class="emotion-0 emotion-1"
  >
    <div>
      <h1
        class="emotion-2 emotion-3"
      >
        <label
          class="emotion-4 emotion-5"
          for="palette-form-input"
        >
          Add the colours from your palette
        </label>
      </h1>
      <input
        autocomplete="off"
        class="emotion-6 emotion-7"
        data-testid="palette-form-input"
        id="palette-form-input"
        name="paletteFormInput"
        placeholder="e.g. #ccc, #fff or blue"
        type="text"
        value=""
      />
    </div>
  </form>
</DocumentFragment>
`;

exports[`PaletteInput should render an error message when passed 1`] = `
<DocumentFragment>
  .emotion-0 {
  text-align: center;
}

.emotion-2 {
  font-weight: 700;
  line-height: 1;
  margin: 3rem 0;
}

.emotion-4 {
  display: inline-block;
  font-size: 2.5rem;
  margin-bottom: 0.5rem;
  width: 100%;
}

.emotion-6 {
  background: rgba(255, 255, 255, 0.1);
  border: solid 1px;
  border-color: #c12915;
  border-radius: 3px;
  color: #c12915;
  font-size: 1.2rem;
  padding: 0.5rem 1rem;
  margin-bottom: 0;
  width: 100%;
}

.emotion-6:focus {
  border-color: #c12915;
  box-shadow: 0px 0px 10px #c12915;
}

.emotion-8 {
  color: #c12915;
  font-size: 1.3rem;
  margin: 1rem 0;
}

<form
    class="emotion-0 emotion-1"
  >
    <div>
      <h1
        class="emotion-2 emotion-3"
      >
        <label
          class="emotion-4 emotion-5"
          for="palette-form-input"
        >
          Add the colours from your palette
        </label>
      </h1>
      <input
        aria-describedby="error-message-label-palette-form-input"
        autocomplete="off"
        class="emotion-6 emotion-7"
        data-testid="palette-form-input"
        id="palette-form-input"
        name="paletteFormInput"
        placeholder="e.g. #ccc, #fff or blue"
        type="text"
        value=""
      />
      <div
        class="emotion-8 emotion-9"
        id="error-message-label-palette-form-input"
      >
        I'm an error
      </div>
    </div>
  </form>
</DocumentFragment>
`;
